app/scripts/metamask-controller.js

```diff
@@ -6916,13 +6943,14 @@ export default class MetamaskController extends EventEmitter {
         // This should be the same as the return value of `eth_accounts`,
         // namely an array of the current / most recently selected Ethereum
         // account.
+        // TODO: Can this be simplified?...
         params:
           newAccounts.length < 2
             ? // If the length is 1 or 0, the accounts are sorted by definition.
               newAccounts
             : // If the length is 2 or greater, we have to execute
               // `eth_accounts` vi this method.
-              await this.getPermittedAccounts(origin),
+              await this.getPermittedAccountsSorted(origin),
```

Comment:
thoughts? unsure of if there are some side effects here that result from an unnecessary await in the case where there is only one new account

Reply:
i don't there would be, but wisdom/experience suggests meddling with simple looking changes like this

Reply:
Do you just mean from a performance stand point?

Reply:
right, sacrificing some performance for simpler code. we could make this 7 line block into 1

I'm leaning towards just leaving it alone though. I have a bad habit of trying to clean up too much nearby code when doing large refactors. I feel like the dad from Malcom In the Middle https://www.youtube.com/watch?v=_UZFI-8D5uA

Reply:
going to remove the comment and mark this as resolved